examples/11_plurigaussian/02_spatial_relations.py

```diff
@@ -0,0 +1,111 @@
+"""
+Controlling Spatial Relations
+-----------------------------
+
+In this example we will try to understand how we can
+influence the spatial relationships of the different
+categories with the L-field. For simplicity, we will
+start very similarly to the very first example.
+"""
+
+import matplotlib.pyplot as plt
+import numpy as np
+
+import gstools as gs
+
+dim = 2
+# no. of cells in both dimensions
+N = [100, 80]
+
+x = np.arange(N[0])
+y = np.arange(N[1])
+
+###############################################################################
+# Again, we will use the same parameters for both fields.
+
+model = gs.Gaussian(dim=dim, var=1, len_scale=10)
+srf = gs.SRF(model)
+field1 = srf.structured([x, y], seed=20170519)
+field2 = srf.structured([x, y], seed=19970221)
+
+###############################################################################
+# Now, we will prepare the L-field, which will be a bit more complicated this
+# time. First, we will create a triangle. Next, we will create two rectangles
+# touching each other along one of their edges and both being directly above the
+# triangle, but without touching it directly. Finally, we will create a few
+# very narrow rectangles, which will not touch any other category shapes. The
+# implementation details are not very interesting, and can be skipped.
+
+# no. of grid cells of L-field
+M = [60, 50]
+
+# size of the rectangles
+R = [10, 8]
+
+# positions of some of the shapes for concise indexing
+S1 = [1, -9]
+S2 = [-5, 3]
+S3 = [-5, -5]
+
+L = np.zeros(M)
+# a small upper triangular helper matrix to create the triangle
+tri = np.triu(np.ones((R[0], R[0])))
+# the triangle
+L[
+    M[0] // 2 + S1[0] : M[0] // 2 + S1[0] + R[0],
+    M[1] // 2 + S1[1] : M[1] // 2 + S1[1] + R[0],
+] = tri
+# the first rectangle
+L[
+    M[0] // 2 + S2[0] - R[0] // 2 : M[0] // 2 + S2[0] + R[0] // 2,
+    M[1] // 2 + S2[1] - R[1] // 2 : M[1] // 2 + S2[1] + R[1] // 2,
+] = 2
+# the second rectangle
+L[
+    M[0] // 2 + S3[0] - R[0] // 2 : M[0] // 2 + S3[0] + R[0] // 2,
+    M[1] // 2 + S3[1] - R[1] // 2 : M[1] // 2 + S3[1] + R[1] // 2,
+] = 3
+# some very narrow rectangles
+for i in range(4):
+    L[
+        M[0] // 2 + S1[0] : M[0] // 2 + S1[0] + R[0],
+        M[1] // 2
+        + S1[1]
+        + R[1]
+        + 3
+        + 2 * i : M[1] // 2
+        + S1[1]
+        + R[1]
+        + 4
+        + 2 * i,
+    ] = (
+        4 + i
+    )
```

Comment:
I don't know if this is common in the literature about PGS, but I really dislike all the single uppercase Letters for matrices used in this context.

I think the whole API and documentation of GSTools is quiet descriptive, or at least tries to be, and this reads almost like Fortran 77 code.

Couldn't we come up with something like a descriptive and self explaining naming convention for these "things"?

Reply:
Ouch, that Fortran 77 comparison hurt my feelings :-D

I found that code pretty ugly already and didn't want to reduce readability by introducing longer variable names. But changed most of the variable names to longer and more descriptive ones, because your criticism is absolutely valid. I didn't change all of them and kept e.g. `M`, similarly as we often use `N` for the number of grid points.
Are you okay with this compromise?